internal/provider/provider.go

```diff
@@ -442,8 +478,46 @@ func providerConfigure(p *schema.Provider) schema.ConfigureContextFunc {
 	}
 }
 
+// buildClient is used to configure behavioral aspects of the provider. To configure the
+// cloud environment and authentication-related settings, use the providerConfigure function.
 func buildClient(ctx context.Context, p *schema.Provider, d *schema.ResourceData, authConfig *auth.Credentials) (*clients.Client, diag.Diagnostics) {
-	skipProviderRegistration := d.Get("skip_provider_registration").(bool)
+	// TODO: Remove this hardcoded default in v4.0
+	providerRegistrations := "legacy"
+	if features.FourPointOhBeta() {
+		providerRegistrations = d.Get("resource_provider_registrations").(string)
+	}
+
+	// TODO: Remove in v5.0
+	if d.Get("skip_provider_registration").(bool) {
+		if providerRegistrations != "legacy" {
+			return nil, diag.Errorf("provider property `%[2]s` cannot be set at the same time as `%[1]s`, please remove `%[2]s` from your configuration or unset the `%[3]s` environment variable", "resource_provider_registrations", "skip_provider_registration", "ARM_SKIP_PROVIDER_REGISTRATION")
```

Comment:
Minor, but is there a need to use string interpolation for the error message here when the variables are hardcoded strings? As a reviewer I'd find it easier to not have to do the correct substitution myself in order to understand the error message that will be surfaced to the user
```suggestion
			return nil, diag.Error("provider property `skip_provider_registration` cannot be set at the same time as `resource_provider_registrations`, please remove `skip_provider_registration` from your configuration or unset the `ARM_SKIP_PROVIDER_REGISTRATION` environment variable")
```

Reply:
Perhaps a case of over-optimization, I'll make this a hardcoded string 👍